### PR TITLE
Use windows CI build

### PR DESCRIPTION
**Pull request by christophfroehlich**
The main purpose is to avoid adding code which does not compile on Windows. Github provides windows-2019 runners, which can be used here. However, https://github.com/nektos/gh-act does not support this and I was only to test the workflow online.

- Binary releases are not published for rolling, so I hardcoded jazzy
- I added it now to be triggered at PRs and pushes, but it might be configured to run on comments/labels only.
- Needs https://github.com/ros-controls/ros2_control_ci/pull/175

### CODE REVIEW

File: .github/workflows/binary-build-win.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  #   types:
+  #     - created
+
+jobs:
+  binary-windows:
+    # if: |
+    #   (github.event_name == 'issue_comment' && contains(github.event.comment.body, '/check-windows')) ||
+    #   (github.event_name == 'pull_request' && contains(github.event.label.name, 'check-windows')) ||
+    #   (github.event_name == 'workflow_dispatch')
+    uses: ros-controls/ros2_control_ci/.github/workflows/reusable-ros-tooling-win-build.yml@windows_ci
```
Comment (christophfroehlich) [inline .github/workflows/binary-build-win.yml:25]:
```suggestion
    uses: ros-controls/ros2_control_ci/.github/workflows/reusable-ros-tooling-win-build.yml@master
```

File: test/test_async_function_handler.hpp [hunk 2]
```diff
@@ -24,9 +24,9 @@
 namespace realtime_tools
 {
 enum class return_type : std::uint8_t {
-  OK = 0,
-  ERROR = 1,
-  DEACTIVATE = 2,
```
Comment (saikishor) [inline test/test_async_function_handler.hpp:29]: Can they be capital as usual?

File: test/test_async_function_handler.cpp [hunk 3]
```diff
@@ -401,7 +401,7 @@ TEST_F(AsyncFunctionHandlerTest, check_exception_handling)
   async_class.get_handler().start_thread();
   trigger_status = async_class.trigger();
   ASSERT_TRUE(trigger_status.first);
-  ASSERT_EQ(realtime_tools::return_type::OK, trigger_status.second);
```
Comment (saikishor) [inline test/test_async_function_handler.cpp:404]: same here and the rest of changes

File: test/test_async_function_handler.hpp (hunk 2 above)
Comment (christophfroehlich) [reply test/test_async_function_handler.hpp:29]: no, see the other comments

**Comment by SENAI-GilmarCorreia** [comment]
Here's the corrected English version:

@christophfroehlich I am trying to understand the problem here. Do you want to revert the enums back to capital letters?

As I have seen, ERROR already exists in Windows systems because this identifier is reserved for error codes in Windows.

You can simply rename it by changing the case, as you did, or you can change the name of the enum:

```cpp
enum class return_type : std::uint8_t {
  OK = 0,
  FAILURE = 1,
  DEACTIVATE = 2,
};
```

I'm testing here as well.

**Comment by christophfroehlich** [comment]
Thanks for reaching out. Exactly, if only the ERROR is a problem then I'd rename it as you suggested. Or maybe there is a compile definition to avoid this, like NOMINMAX ;)

**Comment by SENAI-GilmarCorreia** [comment]
As I've seen, this definition is really important. You can use #undef ERROR, but it might cause some unexpected behaviors.

**Review by SENAI-GilmarCorreia — APPROVED** [review]

**Review by fmauch — APPROVED** [review]
This looks fine with me

**Comment by christophfroehlich** [comment]
@mergifyio backport humble

**Pull request merged**